backend/typescript/services/implementations/donorService.ts

```diff
@@ -148,11 +149,23 @@ class DonorService implements IDonorService {
 
   async deleteDonorById(id: string): Promise<void> {
     try {
-      const deletedRole: Donor | null = await Donor.findByPk(Number(id));
+      const deletedDonor: Donor | null = await Donor.findByPk(Number(id));
 
-      if (!deletedRole) {
+      if (!deletedDonor) {
         throw new Error(`id ${id} not found.`);
       }
+      try {
+        await Scheduling.destroy({
+          where: { donor_id: deletedDonor.id },
+        });
+      } catch (error) {
+        Logger.error(
+          `Failed to delete schedules for the donor. Reason = ${getErrorMessage(
+            error,
+          )}`,
```

Comment:
so we're not rolling back the schedule if the donor deletion is unsuccessful?

Reply:
yeah so if donor deletion is unsuccessful, we are only adding back the donor, not anything associated to it.